pkg/backends/otlp/internal/data/map.go

```diff
@@ -51,9 +51,18 @@ func (m Map) Equal(mm Map) bool {
 	)
 }
 
+func (m Map) Hash() string {
+	var sb strings.Builder
+	for i := 0; i < len(*m.raw); i++ {
+		sb.WriteString((*m.raw)[i].Key)
+		sb.WriteString((*m.raw)[i].Value.String())
+	}
+	return sb.String()
+}
+
 func (m Map) Insert(key string, value string) {
 	index := slices.IndexFunc(*m.raw, func(kv *v1common.KeyValue) bool {
-		return kv.Key == key
+		return kv.Key <= key
```

Comment:
Are we guaranteeing that this is sorted with this logic? If so we should change the naming of the data struct to include that info.